### ISSUE TITLE
[CI] RetrySearchIntegTests testSearcherId failing

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#71887]
I've seen this twice now on Windows. We're getting an `AccessDeniedException` but I wonder if the _real_ problem is us hitting path length limits on Windows.

**Build scan:**
https://gradle-enterprise.elastic.co/s/az3mlzx6obs6e/tests/:x-pack:plugin:searchable-snapshots:internalClusterTest/org.elasticsearch.xpack.searchablesnapshots.RetrySearchIntegTests/testSearcherId

**Reproduction line:**
`gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.RetrySearchIntegTests.testSearcherId" -Dtests.seed=C1C5C5DD639AFC9C -Dtests.security.manager=true -Dtests.locale=es-PY -Dtests.timezone=Pacific/Truk -Druntime.java=11`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.searchablesnapshots.RetrySearchIntegTests&tests.test=testSearcherId

**Failure excerpt:**
```
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=1115, name=elasticsearch[node_s1][generic][T#5], state=RUNNABLE, group=TGRP-RetrySearchIntegTests]

  at __randomizedtesting.SeedInfo.seed([C1C5C5DD639AFC9C:2346010D078D876E]:0)

  Caused by: java.lang.AssertionError: java.io.UncheckedIOException: java.nio.file.AccessDeniedException: C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\x-pack\plugin\searchable-snapshots\build\testrun\internalClusterTest\temp\org.elasticsearch.xpack.searchablesnapshots.RetrySearchIntegTests_C1C5C5DD639AFC9C-001\tempDir-002\node_s1\indices\tGDphX3nTiO2W3sIN_5IDQ\0\snapshot_cache\y4BzhBEPTjWtxQzB9FgToQ\l-wM3LV7R_6aCfijDfCBXw

    at __randomizedtesting.SeedInfo.seed([C1C5C5DD639AFC9C]:0)
    at org.elasticsearch.common.util.concurrent.AbstractRefCounted.decRef(AbstractRefCounted.java:57)
    at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile.decrementRefCount(CacheFile.java:247)
    at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile.startEviction(CacheFile.java:275)
    at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:74)
    at org.elasticsearch.core.internal.io.IOUtils.closeWhileHandlingException(IOUtils.java:164)
    at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.onCacheFileEviction(CacheService.java:485)
    at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.lambda$new$1(CacheService.java:161)
    at org.elasticsearch.common.cache.Cache.delete(Cache.java:788)
    at org.elasticsearch.common.cache.Cache.lambda$new$6(Cache.java:488)
    at org.elasticsearch.common.cache.Cache$CacheSegment.remove(Cache.java:305)
    at org.elasticsearch.common.cache.Cache.invalidate(Cache.java:518)
    at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.processShardEviction(CacheService.java:409)
    at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService$1.doRun(CacheService.java:345)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
    at java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:669)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.lang.Thread.run(Thread.java:834)

    Caused by: java.io.UncheckedIOException: java.nio.file.AccessDeniedException: C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\x-pack\plugin\searchable-snapshots\build\testrun\internalClusterTest\temp\org.elasticsearch.xpack.searchablesnapshots.RetrySearchIntegTests_C1C5C5DD639AFC9C-001\tempDir-002\node_s1\indices\tGDphX3nTiO2W3sIN_5IDQ\0\snapshot_cache\y4BzhBEPTjWtxQzB9FgToQ\l-wM3LV7R_6aCfijDfCBXw

      at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile$1.closeInternal(CacheFile.java:71)
      at org.elasticsearch.common.util.concurrent.AbstractRefCounted.decRef(AbstractRefCounted.java:55)
      at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile.decrementRefCount(CacheFile.java:247)
      at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile.startEviction(CacheFile.java:275)
      at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:74)
      at org.elasticsearch.core.internal.io.IOUtils.closeWhileHandlingException(IOUtils.java:164)
      at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.onCacheFileEviction(CacheService.java:485)
      at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.lambda$new$1(CacheService.java:161)
      at org.elasticsearch.common.cache.Cache.delete(Cache.java:788)
      at org.elasticsearch.common.cache.Cache.lambda$new$6(Cache.java:488)
      at org.elasticsearch.common.cache.Cache$CacheSegment.remove(Cache.java:305)
      at org.elasticsearch.common.cache.Cache.invalidate(Cache.java:518)
      at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.processShardEviction(CacheService.java:409)
      at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService$1.doRun(CacheService.java:345)
      at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
      at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
      at java.util.concurrent.FutureTask.run(FutureTask.java:264)
      at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:669)
      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
      at java.lang.Thread.run(Thread.java:834)

      Caused by: java.nio.file.AccessDeniedException: C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\x-pack\plugin\searchable-snapshots\build\testrun\internalClusterTest\temp\org.elasticsearch.xpack.searchablesnapshots.RetrySearchIntegTests_C1C5C5DD639AFC9C-001\tempDir-002\node_s1\indices\tGDphX3nTiO2W3sIN_5IDQ\0\snapshot_cache\y4BzhBEPTjWtxQzB9FgToQ\l-wM3LV7R_6aCfijDfCBXw

        at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:89)
        at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
        at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
        at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:270)
        at sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
        at org.apache.lucene.mockfile.FilterFileSystemProvider.deleteIfExists(FilterFileSystemProvider.java:238)
        at org.apache.lucene.mockfile.FilterFileSystemProvider.deleteIfExists(FilterFileSystemProvider.java:238)
        at org.apache.lucene.mockfile.FilterFileSystemProvider.deleteIfExists(FilterFileSystemProvider.java:238)
        at org.apache.lucene.mockfile.FilterFileSystemProvider.deleteIfExists(FilterFileSystemProvider.java:238)
        at java.nio.file.Files.deleteIfExists(Files.java:1180)
        at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile$1.closeInternal(CacheFile.java:69)
        at org.elasticsearch.common.util.concurrent.AbstractRefCounted.decRef(AbstractRefCounted.java:55)
        at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile.decrementRefCount(CacheFile.java:247)
        at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile.startEviction(CacheFile.java:275)
        at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:74)
        at org.elasticsearch.core.internal.io.IOUtils.closeWhileHandlingException(IOUtils.java:164)
        at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.onCacheFileEviction(CacheService.java:485)
        at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.lambda$new$1(CacheService.java:161)
        at org.elasticsearch.common.cache.Cache.delete(Cache.java:788)
        at org.elasticsearch.common.cache.Cache.lambda$new$6(Cache.java:488)
        at org.elasticsearch.common.cache.Cache$CacheSegment.remove(Cache.java:305)
        at org.elasticsearch.common.cache.Cache.invalidate(Cache.java:518)
        at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.processShardEviction(CacheService.java:409)
        at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService$1.doRun(CacheService.java:345)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:669)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.lang.Thread.run(Thread.java:834)

```

**Comment by original-brownbear** [comment]
This looks like it has the same underlying issue as https://github.com/elastic/elasticsearch/issues/71824

**Comment by original-brownbear** [comment]
I wonder if this is an infra issue of sorts. The problem here is that we fail to delete a cache file even though we fully decrement its ref count. This could either be due to a bug where we keep using the file after decrementing the ref count to `0` or due to some outside process touching the file.
I couldn't find any failures on non-Windows where `WindowsFS` runs would theoretically surface a bug by checking if a file is open before deleting.

**Comment by mark-vieira** [comment]
This looks similar, any chances it's related?
https://gradle-enterprise.elastic.co/s/oyww26e5bysz2/tests/:x-pack:plugin:searchable-snapshots:internalClusterTest/org.elasticsearch.xpack.searchablesnapshots.FrozenSearchableSnapshotsIntegTests/testCreateAndRestorePartialSearchableSnapshot#1

**Comment by original-brownbear** [comment]
@mark-vieira yea it's the same sorry didn't get around to backporting the test fix in #71901 yet. Will do later today.
